src/components/TableComponent/index.jsx

```diff
@@ -70,7 +68,6 @@ class TableComponent extends React.Component {
       <div className={className}>
         <div className="row">
           <div className="col">
-            {loading && <TableLoadingOverlay />}
```

Comment:
Can y'all double check my logic here? Based on lines 136 and 139, I don't think we can ever get to this state, but I want to be sure I haven't missed anything.

Reply:
The purpose of this component is to display a small loading message overlaying an existing table. The `LoadingMessage` component is rendered first when no data exists at all for the table. When page 1 of a table is rendered, and then the user paginates, we had the `TableLoadingOverlay` component overlay a small loading message on top of the table to give the affordance that the table is reloading, which helped avoid constantly swapping in and out the `LoadingMessage` - or a skeleton, now - when the user paginates which could get annoying for the user:

![image](https://user-images.githubusercontent.com/2828721/116115228-2ddaa900-a688-11eb-8304-55a884c3edfe.png)

So, this line _is_ reachable when the table is `loading` and we already have existing data from a previous render of the table.

Reply:
What's there is clunky. Frankly I'd remove the `&& !data` part of the loading condition at line 133 because to me it looks like it is masking some other error: unless your loading state is just wrong, why would you care about whether data was non-null? That would only happen in a weird race condition.